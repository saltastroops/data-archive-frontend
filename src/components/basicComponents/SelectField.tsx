import * as React from "react";

interface ISelectProps extends React.SelectHTMLAttributes<HTMLSelectElement> {}

/**
 * A select field in a form. The actual select is enclosed in a div element
 * with a "control" and a "select" class (as used in Bulma forms). Class
 * names passed to tghis elemebt are applied to this div element.
 *
 * You can pass the same properties as you may to a select component.
 */
const SelectField = (props: ISelectProps) => {
  const { className, children, ...rest } = props;
  return (
    <div className={`control select ${className || ""}`}>
      <select {...rest}>{children}</select>
    </div>
  );
};

export default SelectField;

/**
 * The "any" option. The key attribute has the value "any" and the value
 * attribute has the value "".
 */
export const AnyOption = () => (
  <option key="all" value="All">
    Any
  </option>
);
