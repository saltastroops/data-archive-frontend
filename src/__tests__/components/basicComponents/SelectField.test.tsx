import { shallow, ShallowWrapper } from "enzyme";
import toJson from "enzyme-to-json";
import * as React from "react";
import SelectField from "../../../components/basicComponents/SelectField";

describe("SelectField", () => {
  let wrapper: ShallowWrapper;

  beforeEach(() => {
    const onChange = () => {
      // Do nothing
    };

    wrapper = shallow(
      <SelectField className="danger is-large" onChange={onChange} value="bbb">
        <option value="aaa">AAA</option>
        <option value="bbb">BBB</option>
        <option value="ccc">CCC</option>
      </SelectField>
    );
  });

  it("should render", () => {
    expect(wrapper).toBeDefined();
  });

  it("should render correctly", () => {
    expect(toJson(wrapper)).toMatchSnapshot();

    const noClassNMame = shallow(
      <SelectField value="aaa">
        <option value="aaa">AAA</option>
      </SelectField>
    );
    expect(toJson(noClassNMame)).toMatchSnapshot();
  });
});
