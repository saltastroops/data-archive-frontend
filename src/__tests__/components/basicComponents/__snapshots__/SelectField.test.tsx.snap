// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectField should render correctly 1`] = `
<div
  className="control "
>
  <div
    className="select danger is-large"
  >
    <select
      onChange={[Function]}
      value="bbb"
    >
      <option
        value="aaa"
      >
        AAA
      </option>
      <option
        value="bbb"
      >
        BBB
      </option>
      <option
        value="ccc"
      >
        CCC
      </option>
    </select>
  </div>
</div>
`;

exports[`SelectField should render correctly 2`] = `
<div
  className="control "
>
  <div
    className="select "
  >
    <select
      value="aaa"
    >
      <option
        value="aaa"
      >
        AAA
      </option>
    </select>
  </div>
</div>
`;
